internal/lru/lru.go

```diff
@@ -34,13 +38,15 @@ type Cache struct {
 	// executed when an entry is purged from the cache.
 	OnEvicted func(key string, value interface{})
 
+	lock  sync.RWMutex
 	ll    *list.List
 	cache map[string]*list.Element
 }
 
 type entry struct {
-	key   string
-	value interface{}
+	key       string
+	value     interface{}
+	timestamp time.Time
```

Comment:
Let's add a comment here that explains what `timestamp` is used for. That is, mention that we eliminate the need to take write-lock for every LRU lookup (needed to keep items ordered by LRU in `ll`) by delaying the update until some time period has passed (1 second in this series). We can do this because most recently accessed items are likely to be at the top of the list anyway.

Don't know if we need to mention this, but the technique is mentioned in the following blog post as "windowed updates":

https://www.openmymind.net/High-Concurrency-LRU-Caching/